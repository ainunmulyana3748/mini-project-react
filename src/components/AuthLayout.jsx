import React from "react";
import { Link, useNavigate } from "react-router-dom";
import { ArrowLeft } from "lucide-react";

const AuthLayout = ({
  title,
  description,
  children,
  footerText,
  footerLink,
  footerLinkText,
}) => {
  const navigate = useNavigate();

  return (
    <div className="min-h-screen flex items-center justify-center">
      <div className="p-6 rounded-xl w-96 space-y-4 shadow-lg bg-white">
        <ArrowLeft
          size={16}
          className="mr-2 cursor-pointer"
          onClick={() => navigate(-1)}
        />
        <h1 className="text-2xl font-semibold">{title}</h1>
        <p className="text-sm text-zinc-400">{description}</p>
        {children}
        {footerText && footerLink && (
          <p className="text-center text-sm text-zinc-400">
            {footerText}{" "}
            <Link to={footerLink} className="text-blue-400 hover:underline">
              {footerLinkText}
            </Link>
          </p>
        )}
      </div>
    </div>
  );
};

export default AuthLayout;
