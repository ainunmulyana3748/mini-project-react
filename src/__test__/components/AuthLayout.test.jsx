import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AuthLayout from "../../components/AuthLayout";

// Mock useNavigate
const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("AuthLayout Component", () => {
  const defaultProps = {
    title: "Login",
    description: "Welcome back",
    children: <div>Login Form</div>,
    footerText: "Don't have an account? ",
    footerLink: "/register",
    footerLinkText: "Register here",
  };

  beforeEach(() => {
    // Clear mock before each test
    mockedUsedNavigate.mockClear();
  });

  test("renders with all props correctly", () => {
    render(
      <BrowserRouter>
        <AuthLayout {...defaultProps} />
      </BrowserRouter>
    );

    expect(screen.getByText(defaultProps.title)).toBeInTheDocument();
    expect(screen.getByText(defaultProps.description)).toBeInTheDocument();
    expect(screen.getByText("Login Form")).toBeInTheDocument();
    expect(screen.getByText(/Don't have an account?/)).toBeInTheDocument();
    expect(screen.getByText("Register here")).toBeInTheDocument();
  });

  test("navigates back when back button is clicked", () => {
    render(
      <BrowserRouter>
        <AuthLayout {...defaultProps} />
      </BrowserRouter>
    );

    const backButton = screen.getByTestId("back-button");
    fireEvent.click(backButton);

    expect(mockedUsedNavigate).toHaveBeenCalledWith(-1);
  });

  test("renders without footer when footer props are not provided", () => {
    const propsWithoutFooter = {
      title: "Login",
      description: "Welcome back",
      children: <div>Login Form</div>,
    };

    render(
      <BrowserRouter>
        <AuthLayout {...propsWithoutFooter} />
      </BrowserRouter>
    );

    expect(
      screen.queryByText(/Don't have an account?/)
    ).not.toBeInTheDocument();
    expect(screen.queryByText("Register here")).not.toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <AuthLayout {...defaultProps} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
