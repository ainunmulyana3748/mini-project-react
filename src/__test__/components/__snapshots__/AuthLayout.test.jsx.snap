// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayout Component matches snapshot 1`] = `
<div>
  <div
    class="min-h-screen flex items-center justify-center"
  >
    <div
      class="p-6 rounded-xl w-96 space-y-4 shadow-lg bg-white"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-arrow-left mr-2 cursor-pointer"
        data-testid="back-button"
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m12 19-7-7 7-7"
        />
        <path
          d="M19 12H5"
        />
      </svg>
      <h1
        class="text-2xl font-semibold"
      >
        Login
      </h1>
      <p
        class="text-sm text-zinc-400"
      >
        Welcome back
      </p>
      <div>
        Login Form
      </div>
      <p
        class="text-center text-sm text-zinc-400"
      >
        Don't have an account? 
        <a
          class="text-blue-400 hover:underline"
          data-discover="true"
          href="/register"
        >
          Register here
        </a>
      </p>
    </div>
  </div>
</div>
`;
